### PR TITLE
Run docker image generation on each Push

### PR DESCRIPTION
**Pull request by Munsio**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      fail-fast: false # Run the whole matrix for maximum information, no matter if we fail in one entry early
+      matrix:
+        os:
+          - 'macOS-latest'
+          - 'ubuntu-latest'
+          - 'windows-latest'
+    uses: ./.github/workflows/pipeline.yml
+    with:
+      os: ${{ matrix.os }}
+    needs: docker
```
Comment (ruiAzevedo19) [inline .github/workflows/ci.yml:29]: I don't have much knowledge about this, but shouldn't the workflow `.github/workflows/test.yml` be called somewhere here?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  test:
+    name: 'Run tests and linting'
+    strategy:
+      fail-fast: false # Run the whole matrix for maximum information, no matter if we fail in one entry early
+      matrix:
+        os:
+          - 'macOS-latest'
+          - 'ubuntu-latest'
+          - 'windows-latest'
+    uses: ./.github/workflows/pipeline.yml
```
Comment (ruiAzevedo19) [inline .github/workflows/ci.yml:26]: Which workflow is this one? Can't find the file

Comment (Munsio) [reply .github/workflows/ci.yml:26]: Missed the renaming.

**Comment by ruiAzevedo19** [comment]
@Munsio LGTM

**Comment by ruiAzevedo19** [comment]
@bauersimon Can you also take a look please?

**Pull request merged**